activerecord/test/cases/adapters/postgresql/schema_test.rb

```diff
@@ -832,3 +832,72 @@ def test_nulls_not_set_is_dumped
     assert_no_match(/nulls_not_distinct/, output)
   end
 end
+
+class SchemaWithListPartitionTableTest < ActiveRecord::PostgreSQLTestCase
+  include SchemaDumpingHelper
+
+  setup do
+    @connection = ActiveRecord::Base.connection
+    @connection.create_table "trains", id: false, options: "PARTITION BY LIST (kind)" do |t|
```

Comment:
I think that you need also to wrap the table creation with `supports_native_partitioning?`, because it will fail otherwise.